lib/kflags/kconfig/config.go

```diff
@@ -281,7 +284,7 @@ func FromFlags(fl *Flags) Modifier {
 	}
 }
 
-func NewConfigAugmenterFromDNS(cs cache.Store, domain string, binary string, mods ...Modifier) (*ConfigAugmenter, error) {
+func NewConfigAugmenterFromDNS(fallback config.Store, cs cache.Store, domain string, binary string, mods ...Modifier) (*ConfigAugmenter, error) {
```

Comment:
fallback here ... is also a key value store, pretty much, just different API and semantics. For caching the old result of resoultion, I'd probably recommend using `cache.Store`? which also has the benefit of saving you from having to change the contract of the function? And bringing in an additional library?

(eg, cache.Store is passed in explicitly with the purpose of caching results of operations)

Reply:
cache.Store works very differently - it does not actually store anything in and of itself. Would need to layer a marshaler on top of it, which the config store already provides.